src/components/HourChart/HourChartWithCounts.jsx

```diff
@@ -3,6 +3,96 @@ import d3 from 'd3';
 
 import { HourChart, CountChart } from '../../components';
 import { sum, average } from '../../utils/math';
+import addComputedProps from '../../hoc/addComputedProps';
+
+
+/**
+ * Filter the data and group it by hour and by date
+ * @param {Object} props the component props
+ * @return {Object} the prepared data { filteredData, dataByHour, dataByDate }
+ */
+function prepareData(props) {
```

Comment:
in other components, we've pushed this kind of stuff into a api level transform. i'm fine with it here - just wondering the difference...